### DISCUSSION TITLE
Manual configuration of Wifi Client

### DISCUSSION THREAD

**Discussion by FIAV1** [RaspAP/raspap-webgui#898] [Ideas]
I was wondering, what about manual configuration of Wifi Client? So that a user could add an AP (SSID and password) manually. This way, a user could add an AP that he knows will be available in a certain moment.

Think about this scenario: I'm in the middle of nowhere, with my Rpi and my phone. Not all the smartphones allow to use wifi and hotspot together, so I want to power on the Rpi (with RaspAP spawning an AP), connect to it with my phone, add SSID and password of my hotspot (that is currently unavailable) and save them. Then, when I disconnect my phone and enable the hotspot, the Rpi will connect to it.

I think this can be useful, because the only other possible way is to preconfigure the Rpi "at home" and this feels like a loss of flexibility to me.

**Comment by svenons** [comment]
Why do you need the Raspberry to give out WiFi if you're in nowhere and have only the phone which will be giving out the hotspot? Who will be connecting to it? Use whatever device you want the WiFi for to set it 🥇

**Reply by FIAV1** [reply]
If I need to connect the raspberry to some not-smart-sensor, so that raspberry will communicate with it, maybe I want to give the raspberry an internet connectivity so that it will connect to some remote server and send data to it. This is one possible scenario.

**Reply by billz** [reply]
wpa_supplicant can be configured with any number of networks and assigned priorities:

```
network={
    ssid="home_network"
    psk="password1"
    priority=1
}
network={
   ssid="another_network"
   psk="password2"
   priority=2
}
```
A higher priority value means a network is more desirable. By default networks have priority 0.
RaspAP currently uses wpa_cli to scan for existing networks, but having both a manual option and the ability to assign priority values are good ideas.
